x-pack/legacy/plugins/apm/server/routes/ui_indices.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as t from 'io-ts';
+import { setupRequest } from '../lib/helpers/setup_request';
+import { createRoute } from './create_route';
+import { listUiIndices } from '../lib/settings/ui_indices/list_ui_indices';
+import { getUiIndex } from '../lib/settings/ui_indices/get_ui_index';
+import { saveUiIndices } from '../lib/settings/ui_indices/save_ui_indices';
+
+// get list of ui indices and values
+export const uiIndicesRoute = createRoute(core => ({
+  method: 'GET',
+  path: '/api/apm/settings/ui-indices',
+  handler: async req => {
+    const { server } = core.http;
+    const setup = await setupRequest(req);
+    return await listUiIndices({ setup, server });
+  }
+}));
+
+// get configured APM index
+export const uiIndexRoute = createRoute(core => ({
+  method: 'GET',
+  path: '/api/apm/settings/ui-indices/{indexConfigurationName}',
+  params: {
+    path: t.type({
+      indexConfigurationName: t.string
+    })
+  },
+  handler: async (req, { path }) => {
+    const { server } = core.http;
+    const setup = await setupRequest(req);
+    const { indexConfigurationName } = path;
+    return await getUiIndex({ setup, server, indexConfigurationName });
```

Comment:
Instead of filtering on the backend, what about simply returning all config options:

```ts
return await getApmIndices(server);
```

It's not a big response, so the savings ar negligible.

Reply:
This way you can get rid of `get_ui_index.ts`